CodePush.m

```diff
@@ -27,13 +27,29 @@ + (NSURL *)getBundleUrl
 {
     NSError *error;
     NSString *packageFolder = [CodePushPackage getCurrentPackageFolderPath:&error];
+    NSURL *jsCodeLocation = [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
     
     if (error || !packageFolder)
     {
-        return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
+        return jsCodeLocation;
     }
     
     NSString *packageFile = [packageFolder stringByAppendingPathComponent:UpdateBundleFileName];
+    
+    NSDictionary *mainFileAttribs = [[NSFileManager defaultManager] attributesOfItemAtPath:[jsCodeLocation absoluteString] error:nil];
+    NSDictionary *appFileAttribs = [[NSFileManager defaultManager] attributesOfItemAtPath:packageFile error:nil];
+
+    NSDate *mainDate = [mainFileAttribs objectForKey:NSFileCreationDate];
+    NSDate *appDate = [appFileAttribs objectForKey:NSFileCreationDate];
+    NSComparisonResult result = [mainDate compare:appDate];
+    switch(result) {
+        case NSOrderedSame: // 一致したとき
+        case NSOrderedAscending: // date1が小さいとき
+            return [[NSURL alloc] initFileURLWithPath:packageFile];
+            
+        case NSOrderedDescending: // date1が大きいとき
+            return jsCodeLocation;
+    }
     return [[NSURL alloc] initFileURLWithPath:packageFile];
```

Comment:
If you do want to merge this, I would replace the switch-case with an if-else, that way we get rid of the last return statement and make the code look cleaner.

```objectivec
if (result == NSOrderedSame || result == NSOrderedAscending) {
    return [[NSURL alloc] initFileURLWithPath:packageFile];
} else {
    return jsCodeLocation;
}